### ISSUE TITLE
Font renderer

### ISSUE THREAD

**Issue by watzon** [johnjansen/crystal-libraries-needed#111]
This would be a nice thing to have which could augment the abilities of the [stumpy](https://github.com/stumpycr/stumpy_core) libraries. Currently there's no way to add text to images, but with a font renderer it would be pretty easy to add that ability.

**Comment by bew** [comment]
Something like #55 or harfbuzz would be great :)

**Comment by watzon** [comment]
@bew agreed, I remember seeing harfbuzz recently as a suggested repo from github. A wrapper around it would be awesome, although it would also be pretty cool to have a pure Crystal font renderer as well.

**Comment by bew** [comment]
In stumpy_utils, there is a way to write text, but it's pretty basic compared to the mentioned libs above

https://github.com/stumpycr/stumpy_utils/blob/master/README.md#canvastextx-baseline_y-text-font--pcfparserfont-color--rgbaback

**Comment by watzon** [comment]
That's cool, I didn't even realize that. Never actually heard of the `pcf` format.

**Comment by bew** [comment]
From what I read, text rendering [is a hard topic](http://behdad.org/text/), and I think it would be simpler to wrap a library that does it well instead of re-inventing such a complex wheel..

Also, I read that HarfBuzz is a text shaping engine, and Pango uses it for the display/placement of text, so wrapping Pango would be the best option here (and it's cross-plateform!).

**Comment by watzon** [comment]
Sounds good to me! Now someone just needs to do it :grin:

**Comment by hugopl** [comment]
You can acess Pango and/or Harfbuzz using https://github.com/jhass/crystal-gobject, i.e. problem solved. :-)
